cmd/sentinel/cmd/sentinel.go

```diff
@@ -974,7 +974,7 @@ func (s *Sentinel) updateCluster(cd *cluster.ClusterData, pis cluster.ProxiesInf
 
 				// get db associated to the keeper
 				db := cd.FindDB(k)
-				if db != nil {
+				if db != nil && s.dbType(newcd, db.UID) != dbTypeMaster {
```

Comment:
I think we will want to ensure that both the keeper and db remain intact in the event it's master.